src/sparseml/transformers/compression/README.md

```diff
@@ -30,52 +30,131 @@ needed for decompression in the compressed state_dict:
 }
 ```
 
+Config information gets stored in the HF config file
+```json
+// config.json
+{
+    "sparsity_config": {
+        "format": "sparse_bitmask", // "dense_sparsity" for original tensor format
+
+        // informational
+        "sparsity_structure": "unstructured", // or 2:4, 8:16 etc...
+        "global_sparsity": "0.5"
+    }
+}
+```
+
+## Saving/Loading Interface 
+
+Loading in a compressed model requires no interface changes
+
+```python
+from sparseml.transformers.utils import SparseAutoModelForCausalLM
+
+# should contain model.safetensors or model.safetensors.index.json
+model_path = "/PATH/TO/COMPRESSED_MODEL"
+
+model = SparseAutoModelForCausalLM.from_pretrained(
+    model_name_or_path=model_path,
+    **model_kwargs,
+)
+```
+
+Saving a compressed model with an explicitly provided compression config. The config
+is saved to the model's `config.json` file. **Note:** the model must have been 
+initialized with SparseAutoModelForCausalLM.from_pretrained()
+
+```python
+from sparseml.transformers.compression import BitmaskConfig
+
+output_dir = "/PATH/TO/SAVE/COMPRESSED_MODEL"
+sparsity_config = BitmaskConfig()
+
+model.save_pretrained(
+    save_directory=output_dir,
+    sparsity_config=sparsity_config,
+)
+```
+
+Saving a compressed model, inferring the config from the model attributes
+
+```python
+model.save_compressed(
+    model,
+    save_directory=output_dir,
+)
+
+# alternative
+model.save_pretrained(
```

Comment:
I think having multiple ways of doing something is confusing

So let's eliminate any external reference to `model.save_compressed` (if not too much problem to remove internally that would be great too) and standardize on:

```python
model.save_pretrained(
   save_directory="output_dir",
   save_compressed=True
)

Reply:
It would be simple to remove internally, I'm not tied to keeping it but it was requested by @mgoin to add this in. GPTQ has a similar `model.save_quantized`

Reply:
@mgoin can you clarify why we need `save_compressed`?

I don't see what advantage there is to having multiple pathways to do the same thing

Reply:
It is simply that I like the AutoGPTQ `save_quantized` that is a thin wrapper around `save_pretrained` with the right arg. If you think it is confusing then I'm not deeply tied to it and it can be removed